blockchain_storage/src/store.rs

```diff
@@ -416,9 +439,9 @@ mod test {
                 })
             };
             let thr1 = {
-                let mut store = store.clone();
+                let store = Store::clone(&store);
                 common::thread::spawn(move || {
-                    let tx_result = store.transaction(|tx| {
+                    let tx_result = store.write(|tx| {
```

Comment:
I'm not a big fan of using `write` as a member function name for this... it's a little confusing.

Reply:
I am open to suggestions. I was following the naming scheme from `std::sync::RwLock` but `transaction_{ro,rw}` sounds sensible to me as well

Reply:
`transaction_{ro,rw}` sounds good to me. And to be honest, I'd prefer calling this "inline transactions", but maybe `inline_transaction_{ro,rw}` is an overkill?

Reply:
To be honest, I don't like the `inline_transaction_x` suggestion after thinking about that for a bit. Currently my preferred variant is `start_transaction_x` and `transaction_x`. I find that fairly descriptive: `start_transaction_x` starts a transaction, `transaction_x` defines the whole transaction. That is merely a different interface to the same thing, not a different "kind" of transaction, as the `inline_` prefix seems to suggest.

Reply:
That's OK. Use what you suggested.